File: rust-numeric/src/sfuns.rs
use std::f64::consts::PI;

const ERF_A: [f64; 5] = [
    0.254829592,
    -0.284496736,
    1.421413741,
    -1.453152027,
    1.061405429,
];

pub fn erf(x: f64) -> f64 {
    if x < (0.into()) {
        -erf(-x)
    } else {
        let t = 1. / (1. + 0.3275911 * x);
        let sum = t * (ERF_A[0] + t * (ERF_A[1] + t * (ERF_A[2] + t * (ERF_A[3] + t * ERF_A[4])))); /* the right thing */
        1. - sum * (-x * x).exp()
    }
}

pub fn fgamma(x: f64) -> f64 {
    // single precision gamma function (Gergo Nemes, from Wikipedia)
    if x < 0. {
        PI / (PI * x).sin() / fgamma(1. - x)
    } else if x < 9. {
        fgamma(x + 1.) / x
    } else {
        let lngamma = x * (x + 1. / (12. * x - 1. / x / 10.)).ln() - x + (2. * PI / x).ln() / 2.;
        lngamma.exp()
    }
}

File: exercises/math/sfuns.cs
using static System.Math;
public static class sfuns {
    public static double fgamma(double x) {
        ///single precision gamma function (formula from Wikipedia)
        if(x<0) return PI/Sin(PI*x)/fgamma(1-x); // Euler's reflection formula
        if(x<9) return fgamma(x+1)/x; // Recurrence relation
        double lnfgamma=x*Log(x+1/(12*x-1/x/10))-x+Log(2*PI/x)/2;
        return Exp(lnfgamma);
    }
    public static double lngamma(double x) {
        ///single precision gamma function (formula from Wikipedia)
        if(x<0) return double.NaN;
        if(x<9) return lngamma(x+1) - Log(x); // log(gamma(x-1)/x)
        double lnfgamma=x*Log(x+1/(12*x-1/x/10))-x+Log(2*PI/x)/2;
        return lnfgamma;
    } 
    public static int Fact(int x) {
        // Does C# not have a factorial function?
        if(x<=1) return 1;
        return Fact(x-1)*x;
    }
}
